openmls/src/group/public_group/validation.rs

```diff
@@ -488,29 +484,12 @@ impl PublicGroup {
             return Err(ExternalCommitValidationError::InvalidInlineProposals);
         }
 
-        let remove_proposals = proposal_queue.filtered_by_type(ProposalType::Remove);
-        for proposal in remove_proposals {
-            if proposal.proposal_or_ref_type() == ProposalOrRefType::Proposal {
-                if let Proposal::Remove(remove_proposal) = proposal.proposal() {
-                    let removed_leaf = remove_proposal.removed();
-
-                    if let Some(new_leaf) = path_leaf_node {
-                        // ValSem243: External Commit, inline Remove Proposal:
-                        //            The identity and the endpoint_id of the
-                        //            removed leaf are identical to the ones
-                        //            in the path leaf node.
-                        let removed_leaf = self
-                            .treesync()
-                            .leaf(removed_leaf)
-                            .ok_or(ExternalCommitValidationError::UnknownMemberRemoval)?;
-                        if removed_leaf.credential().identity() != new_leaf.credential().identity()
-                        {
-                            return Err(ExternalCommitValidationError::InvalidRemoveProposal);
-                        }
-                    };
-                }
-            }
-        }
+        // If a Remove proposal is present,
+        // the credential in the LeafNode MUST present a set of
+        // identifiers that is acceptable to the application for
+        // the removed participant.
+        // This MUST be checked by the application.
```

Comment:
Is there an issue that we can link here? Or maybe just add this to the book?

Reply:
Right, so this is a case where the application is responsible for the check and we can't do it (anymore). I added a sentence to the book and started a section. We'll want to fill that section more.